release/create_candidate_pr.py

```diff
@@ -113,24 +115,27 @@ def create_candidate_main(raw_args):
             "Couldn't find a recent commit without test failures. Aborting"
         )
 
-    branch_name = utils.rc_branch_name_for_date(args.release_date.date())
+    branch_name = RELEASE_CANDIDATE_BRANCH
 
     logger.info(
         "Branching {rc} off {sha}. ({msg})".format(
             rc=branch_name, sha=commit_hash, msg=message
         )
     )
     try:
+        github_api.delete_branch(branch_name)
         github_api.create_branch(branch_name, commit_hash)
     except RequestFailed:
-        logger.error("Unable to create branch. Aborting")
+        logger.error("Unable to recreate branch {branch_name}. Aborting"
+                     .format(branch_name=branch_name))
         raise
 
     logger.info(
         "Creating Pull Request for {rc} into release".format(rc=branch_name)
     )
     try:
-        request_title = "Release Candidate {rc}".format(rc=branch_name)
+        request_title = "Release Candidate {rc}"\
+            .format(rc=utils.rc_branch_name_for_date(args.release_date.date()))
```

Comment:
Nit: As releases increase in frequency due to automation, this naming might need to be more fine-grained. But - everything would still work if two releases were performed in a day.